[status-im/status-react#9942] relative paths

rasom opened the pull request:
Storing absolute path for different configs breaks compatibility on iOS as app's dir is changed after upgrade. The solution is to store relative paths and to concatenate it with `backend.rootDataDir`. The only exception is `LogFile` as it is stored outside `backend.rootDataDir` on Android. `LogDir` config was added to allow adding of custom dir for log file.
Configs concerned:
`DataDir`
`LogDir`
`LogFile`
`KeystoreDir`
`BackupDisabledDataDir`

status: ready

ghost commented:
<!--prchecklist--> 
### Pull Request Checklist
- [ ] Have you updated the documentation, if impacted (e.g. [docs.status.im](https://docs.status.im/docs/build_status_go.html))?

status-im-auto commented:
### Jenkins Builds
<details>
<summary>Click to see older builds (27)</summary>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 4934ccab | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-1825/1/) | 2020-01-31 12:42:47 | ~52 sec | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-200131-124154-4934cc.zip) |
| :heavy_check_mark: | 4934ccab | [#1](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-1825/1/) | 2020-01-31 12:47:22 | ~5 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-200131-124153-4934cc.zip) |
| :heavy_check_mark: | 4934ccab | [#1](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-1825/1/) | 2020-01-31 12:49:10 | ~7 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-200131-124152-4934cc.aar) |
| | | | | | | |
| :heavy_check_mark: | 3787d160 | [#2](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-1825/2/) | 2020-01-31 14:23:48 | ~37 sec | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-200131-142310-3787d1.zip) |
| :x: | 3787d160 | [#2](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-1825/2/) | 2020-01-31 14:24:18 | ~1 min | `ios` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-1825/2/consoleText) |
| :x: | 3787d160 | [#2](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-1825/2/) | 2020-01-31 14:24:50 | ~1 min | `android` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-1825/2/consoleText) |
| | | | | | | |
| :heavy_check_mark: | 6050c8d9 | [#3](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-1825/3/) | 2020-01-31 14:36:35 | ~50 sec | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-200131-143545-6050c8.zip) |
| :heavy_check_mark: | 6050c8d9 | [#3](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-1825/3/) | 2020-01-31 14:38:30 | ~2 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-200131-143544-6050c8.zip) |
| :heavy_check_mark: | 6050c8d9 | [#3](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-1825/3/) | 2020-01-31 14:41:19 | ~5 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-200131-143542-6050c8.aar) |
| | | | | | | |
| :heavy_check_mark: | 7b0272c9 | [#4](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-1825/4/) | 2020-01-31 15:41:52 | ~37 sec | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-200131-154114-7b0272.zip) |
| :heavy_check_mark: | 7b0272c9 | [#4](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-1825/4/) | 2020-01-31 15:43:55 | ~2 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-200131-154113-7b0272.zip) |
| :heavy_check_mark: | 7b0272c9 | [#4](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-1825/4/) | 2020-01-31 15:46:55 | ~5 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-200131-154111-7b0272.aar) |
| | | | | | | |
| :heavy_check_mark: | 992168b1 | [#5](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-1825/5/) | 2020-01-31 15:55:14 | ~32 sec | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-200131-155441-992168.zip) |
| :heavy_check_mark: | 992168b1 | [#5](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-1825/5/) | 2020-01-31 15:57:04 | ~2 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-200131-155440-992168.zip) |
| :heavy_check_mark: | 992168b1 | [#5](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-1825/5/) | 2020-01-31 16:00:04 | ~5 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-200131-155438-992168.aar) |
| | | | | | | |
| :heavy_check_mark: | ae9c559a | [#6](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-1825/6/) | 2020-02-03 09:21:25 | ~42 sec | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-200203-092042-ae9c55.zip) |
| :heavy_check_mark: | ae9c559a | [#6](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-1825/6/) | 2020-02-03 09:23:13 | ~2 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-200203-092041-ae9c55.zip) |
| :heavy_check_mark: | ae9c559a | [#6](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-1825/6/) | 2020-02-03 09:26:30 | ~5 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-200203-092040-ae9c55.aar) |
| | | | | | | |
| :heavy_check_mark: | 4f8be72b | [#7](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-1825/7/) | 2020-02-03 10:34:31 | ~40 sec | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-200203-103351-4f8be7.zip) |
| :heavy_check_mark: | 4f8be72b | [#7](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-1825/7/) | 2020-02-03 10:36:41 | ~2 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-200203-103350-4f8be7.zip) |
| :heavy_check_mark: | 4f8be72b | [#7](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-1825/7/) | 2020-02-03 10:40:41 | ~6 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-200203-103348-4f8be7.aar) |
| | | | | | | |
| :heavy_check_mark: | 40b8164d | [#8](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-1825/8/) | 2020-02-03 13:26:34 | ~49 sec | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-200203-132545-40b816.zip) |
| :heavy_check_mark: | 40b8164d | [#8](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-1825/8/) | 2020-02-03 13:28:03 | ~2 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-200203-132544-40b816.zip) |
| :heavy_check_mark: | 40b8164d | [#8](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-1825/8/) | 2020-02-03 13:31:15 | ~5 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-200203-132543-40b816.aar) |
| | | | | | | |
| :heavy_check_mark: | fa0e2d8a | [#9](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-1825/9/) | 2020-02-03 14:10:18 | ~47 sec | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-200203-140930-fa0e2d.zip) |
| :heavy_check_mark: | fa0e2d8a | [#9](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-1825/9/) | 2020-02-03 14:11:56 | ~2 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-200203-140929-fa0e2d.zip) |
| :heavy_check_mark: | fa0e2d8a | [#9](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-1825/9/) | 2020-02-03 14:15:23 | ~5 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-200203-140928-fa0e2d.aar) |
</details>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | db473803 | [#10](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-1825/10/) | 2020-02-03 17:48:31 | ~41 sec | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-200203-174749-db4738.zip) |
| :heavy_check_mark: | db473803 | [#10](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-1825/10/) | 2020-02-03 17:50:22 | ~2 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-200203-174748-db4738.zip) |
| :heavy_check_mark: | db473803 | [#10](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-1825/10/) | 2020-02-03 17:53:27 | ~5 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-200203-174747-db4738.aar) |
| | | | | | | |
| :heavy_check_mark: | 9cf64084 | [#11](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-1825/11/) | 2020-02-03 18:03:05 | ~33 sec | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-200203-180231-9cf640.zip) |
| :heavy_check_mark: | 9cf64084 | [#11](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-1825/11/) | 2020-02-03 18:05:10 | ~2 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-200203-180230-9cf640.zip) |
| :heavy_check_mark: | 9cf64084 | [#11](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-1825/11/) | 2020-02-03 18:07:53 | ~5 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-200203-180229-9cf640.aar) |

rasom committed:
[status-im/status-react#9942] Upgradable paths in configs

Storing absolute path for different configs breaks compatibility on iOS
as app's dir is changed after upgrade. The solution is to store relative
paths and to concatenate it with `backend.rootDataDir`. The only
exception is `LogFile` as it is stored outside `backend.rootDataDir` on
Android. `LogDir` config was added to allow adding of custom dir for log
file.
Configs concerned:
`DataDir`
`LogDir`
`LogFile`
`KeystoreDir`
`BackupDisabledDataDir`
